### DISCUSSION TITLE
Enroll in Carpentries Workbench beta

### DISCUSSION THREAD

**Discussion by fmichonneau** [datacarpentry/R-ecology-lesson#799] [General]
Hello @datacarpentry/r-ecology-lesson-maintainers 

@zkamvar is [testing out the beta version](https://carpentries.org/blog/2022/05/workbench-beta/) of the Carpentries Workbench and I'd like this lesson to go through the beta testing. Does anyone have any objections?

**Comment by mondorescue** [comment]
I have no objection.

-Brian

**Comment by Teebusch** [comment]
No objections from me either.

**Comment by zkamvar** [comment]
Hi all, I will be making the transformation of today's snapshot and inviting you to collaborate on the pre-beta repository in a few hours. 

I will also be creating a pull request here that will add a banner giving a link for instructors and learners to test out the pre-beta version.

During the 28 day window of the pre-beta (ending on 09 August), please make sure to resolve or close all pull requests as they will no longer be relevant with the new structure.

**Reply by zkamvar** [reply]
I should also note: I just discovered a bug in the parser (https://github.com/data-lessons/lesson-transition/issues/8) that is due to the limitations the authors were facing when writing the episode. Code blocks like this:



````markdown
```{text_answer, ....}
Text that is displayed as an answer block
```
````
did not display in the rendered document because I had not known about them (I only knew that blocks with `answer=TRUE` were parsed as answers). If this is a blocker, I will delay the release by a week.

**Reply by fmichonneau** [reply]
Personally, I do not feel that it's a blocker. There are not many of these blocks and the new syntax with the workbench will make it easy to re-add them manually if needed.

**Comment by zkamvar** [comment]
Hello R Ecology Lesson maintainers, thank you for your patience during the last quarter. I am pleased to announce that the **transition to beta phase for this lesson will take place in two weeks on 2022-10-17**.

## Timing

the beta phase is divided into the following schedule of stages

 - 2022-10-17: pre-beta stage (two repos, two sites)
 - 2022-11-14: beta stage (one repo, two sites)
 - 2023-01-23: pre-release stage (one repo, one site) 

Please let me know if this does not work for you and I will do my best to accommodate. 
Importantly: **please resolve (close or merge) any open pull requests before 2022-11-14** because the beta stage will involve changing the folder structure of this repository which will render any existing pull requests invalid.

## Pre Beta Tasks

During this first pre-beta stage of the Beta Phase, I will give you access to https://github.com/fishtree-attempt/R-Ecology-lesson and encourage you to test out the following things on the lesson to get familiar with The Workbench. Here are some suggestions:

 - clone the lesson and build it locally
 - (optionally trigger and) merge and automated workflow update PRs that @carpentries-bot opens
 - review a PR that includes some failing accessibility checks
 - (R lessons only) review a PR using the preview of rendered changes
 - fix a simple typo in a lesson - perhaps in a relatively non-obvious location e.g. in the setup instructions
 - write a link from one episode to another in the same lesson 
 
I will be recording a short video describing the changes you will see in the next week, so be sure to keep your eyes out for that.

**Comment by zkamvar** [comment]
Hi all, I've opened https://github.com/datacarpentry/R-ecology-lesson/pull/819, so please be sure to merge it on Monday.

**Comment by zkamvar** [comment]
Hi all, 

Now that #819 has merged, I have created the pre-beta snapshot of the lesson at https://github.com/fishtree-attempt/R-ecology-lesson and have invited all the maintainers on this lesson to a team that has maintain access for the lesson. This version of the lesson is a snapshot and is available for you to play around with and test out without risk of affecting the lesson as a whole. In 28 days, I will perform the full conversion on this lesson.  

As a reminder, you can [check out the setup instructions](https://carpentries.github.io/sandpaper-docs/) to get yourself ready to build it locally. 

Here are some tasks that you can try out during this 28 day period and please take a minute to give some feedback: https://carpentries.typeform.com/to/KRBl4IZM

- clone the lesson and build it locally
- merge one of those automated workflow update PRs that is opened by @carpentries-bot
- review a PR that includes some failing accessibility checks
- (R lessons only) review a PR using the preview of rendered changes
- create an inline instructor note
- fix a simple typo in a lesson - perhaps in a relatively non-obvious location e.g. in the setup instructions
- fix some broken syntax in a lesson
- write a link from one episode to another in the same lesson

If you have any questions, feel free to ask!

**Comment by zkamvar** [comment]
Hello @datacarpentry/r-ecology-lesson-maintainers,

As we approach the two-week mark of the pre-beta stage, I want to remind you to play around with the snapshot of your lesson on https://github.com/fishtree-attempt/R-ecology-lesson/. 

Remember that during the pre-beta stage, all changes are temporary, so you can make changes without worry. One suggestion I have for you is to [move some chapter-specific instructor notes](https://github.com/fishtree-attempt/R-ecology-lesson/blob/main/instructors/instructor-notes.md#narrative) to be inline with each chapter. 

As always, if you have any questions, please feel free to contact me or ask here.

**Reply by zkamvar** [reply]
Additionally, when you try it out, please take a minute to [provide feedback](https://carpentries.typeform.com/to/KRBl4IZM) about your experience with the workbench

**Comment by zkamvar** [comment]
Hi @datacarpentry/r-ecology-lesson-maintainers,

We are coming up on one week until we enter the beta stage of the Workbench beta phase, during which this repository will be irreversibly altered. 

**Please note that if you are not yet prepared for this change, we can delay the transition**

Before the transition, however, _all of the [current pull requests](https://github.com/datacarpentry/R-ecology-lesson/pulls) must be resolved_. 

I also _highly_ recommend for you to play around with [the snapshot repository](https://github.com/fishtree-attempt/R-ecology-lesson)! You all have full maintain access and are free to do whatever you like. Here are some options:

- Check out the [automated pull request to update packages](https://github.com/fishtree-attempt/R-ecology-lesson/pull/1) on the snapshot repository. I've written some [documentation of what to expect](https://carpentries.github.io/sandpaper-docs/pull-request). 
- Trigger an [automated pull request to update workflows](https://github.com/fishtree-attempt/R-ecology-lesson/actions/workflows/update-workflows.yaml) (these were recently updated in sandpaper 0.10.7)
- Use `sandpaper::validate_lesson()` locally to find and improve inaccessible elements of the lesson 
- Add an instructor note or two inline for the lesson from [the narrative instructor notes](https://preview.carpentries.org/R-ecology-lesson/instructor/instructor-notes.html#narrative)

As always, please [provide feedback](https://carpentries.typeform.com/to/KRBl4IZM) about your experience with the workbench.

**Comment by zkamvar** [comment]
Hello again @datacarpentry/r-ecology-lesson-maintainers,

I have not yet heard back from any of the maintainers on this lesson since the beginning of the pre-beta stage, so **I am going to put the transition of this lesson to [the beta stage](https://carpentries.github.io/workbench/beta-phase.html#beta) on hold.**

**Please reply here when you feel you are ready to go on with the beta phase**

The maintainer community is comprised of volunteers and the most important thing to me is to protect your time. Everyone has their day jobs and their families to think about. I want to make sure everyone is feeling comfortable with working with The Workbench and comfortable with these changes; rushing the changes will not accomplish that.

Because [the next step of the beta phase](https://carpentries.github.io/workbench/beta-phase.html#beta) involves the irreversible transformation of this repository, I can not go through with it unless I know that you are comfortable with the change and can work with the workbench itself. 

In order to move forward with the beta stage, the following must be addressed:

 - [all open pull requests must be resolved (either merged or closed)](https://github.com/datacarpentry/R-ecology-lesson/pulls)
 - The automated pull requests from The Workbench should be addressed:
   - workflow updates: https://github.com/fishtree-attempt/R-ecology-lesson/pull/2
   - package updates: https://github.com/fishtree-attempt/R-ecology-lesson/pull/1

Thank you all again for being part of this work!

**Comment by zkamvar** [comment]
## Workbench Beta Phase Timeframe Adjustment

> Note: this is identical to the message I sent directly via email

The beta phase has been in progress since the end of October, but **because of various changing circumstances, I will be pushing back the beta phase timeline according to the revised schedule below**, synchronising all of the beta phase repositories. With this new timeline, you will have more time to explore the repository snapshots, I will have time to provide more clear communications, and—because everyone will go through each stage synchronously—you will all be able to exchange tips and tricks together. 

Note that Toby Hodges, Vini Salazar, and I will be hosting weekly helpdesk sessions, whose times we will send to you soon. 

Here is the revised schedule, which you can also find [on The Workbench beta phase website](https://carpentries.github.io/workbench/beta-phase.html):

 - [2022-12-12 (pre-beta)](https://carpentries.github.io/workbench/beta-phase.html#pre-beta): (re-)release all beta phase lessons to the pre-beta stage, which will have a parallel snapshot at https://github.com/fishtree-attempt/ hosting the Workbench preview on https://preview.carpentries.org/. 
    - During this time, we encourage you to explore the repository. Here is a [list of tasks to get started](https://carpentries.github.io/workbench/beta-phase.html#pre-beta-tasks). Remember, This preview version is temporary, so you do not have to worry about making any mistakes here. 
    - Take the time to merge or resolve any open PRs. And yes, this is a big task and it is helpful to [remember how to say 'no'](https://opensource.guide/best-practices/#learning-to-say-no) in a way that respects the contributor’s time and effort. If you feel like you need help with a particular PR, please contact Zhian and he will help craft a response.
 - [2023-02-06 (beta)](https://carpentries.github.io/workbench/beta-phase.html#beta): do in-place conversion of all beta phase lessons to use The Workbench by default. The default lesson site will still point to a snapshot of the jekyll version from 2023-02-06, and the preview lesson site will show the Workbench hosted on the main branch of the repository. 
    - IMPORTANT: before this happens, all maintainers MUST be aware of this change and must merge or resolve any open PRs. Remember that it’s okay to say no and/or contact Zhian for wordsmithing.
    - Zhian will create a video describing what this process is doing
    - The broader maintainer community will be alerted that they have three months to merge or resolve their open PRs at this time.
 - [2023-04-03 (pre-release)](https://carpentries.github.io/workbench/beta-phase.html#pre-release): The Workbench will become the default version of the lesson and the Jekyll version will be removed. A blog post will go out alerting everyone of the change. 


**We will make the conversion of all of our lessons to use The Workbench on 2023-05-01**


Thank you once again for agreeing to participate in the beta phase of The Carpentries Workbench. Your efforts in testing the infrastructure on your established lessons are invaluable to improving the workbench and I want to make sure that we take the time to get that right.

**Comment by zkamvar** [comment]
Hello and happy new year! 

This is a reminder that on **06 February 2023 (just under four weeks away), your lesson repositories will be transformed to use The Carpentries Workbench.** This is the irreversible [second stage of the beta phase](https://carpentries.github.io/workbench/beta-phase.html#beta) where the Workbench version of the lesson will be live and the styles version will be frozen.

### Before 06 February 2023

1. **(MUST DO)** Merge or resolve any open PRs (If you are unsure of how to resolve some PRs that need more time, [here's a guide on how to say 'no'](https://opensource.guide/best-practices/#learning-to-say-no))
2. **(recommended)** Try out the [pre-beta tasks to get familiar with The Workbench](https://carpentries.github.io/workbench/beta-phase.html#pre-beta)

### After 06 February 2023

The lesson repository will change with a trimmed-down git history that reflects authorship. **To prevent accidental overwriting of repository content, I will temporarily change your access to the repository as ‘read-only’.** To regain ‘maintain’ access,

1. **(MUST DO)** [make a fresh clone of or update](https://carpentries.github.io/workbench/beta-phase.html#beta-tasks) your lesson on your computer.
2. **(MUST DO)** respond here with `I am ready for write access :rocket:`

### Have Questions?

If you have any questions, would like help resolving PRs, or anything at all, you can email, slack, or schedule a meeting with me: https://calendly.com/zkamvar.
